### PR TITLE
Migrate identity column to integration tests in Delta

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Follows https://trino.io/docs/current/develop/tests.html#avoid-product-tests

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java [hunk 1]
```diff
@@ -981,6 +981,54 @@ public void testIdentityColumns()
...
+    @Test
+    public void testIdentityColumnTableFeature()
+            throws Exception
+    {
+        String tableName = "test_identity_columns_table_feature_" + randomNameSuffix();
+        Path tableLocation = catalogDir.resolve(tableName);
+        copyDirectoryContents(new File(Resources.getResource("databricks133/identity_columns_table_feature").toURI()).toPath(), tableLocation);
+        assertUpdate("CALL system.register_table(CURRENT_SCHEMA, '%s', '%s')".formatted(tableName, tableLocation.toUri()));
+
+        // Disallowing all statements just in case though some statements may not unrelated to identity columns
```
Comment (pajaks) [inline plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:1017]: Is reading from table omitted intentionally (present in product test)?

Comment (SemionPar) [reply plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeBasic.java:1017]: `testIdentityColumns` asserts read behavior, I think that was the reasoning

**Review by SemionPar — APPROVED** [review]
LGMT % nit

**Review by findinpath — APPROVED** [review]

**Review by pajaks — APPROVED** [review]

**Comment by mosabua** [comment]
I assume you will run with this @ebyhr

**Review by raunaqmorarka — APPROVED** [review]

**Pull request merged**